lib/net.js

```diff
@@ -59,47 +58,57 @@ exports.createServer = function(options, connectionListener) {
 // connect(port, [host], [cb])
 // connect(path, [cb]);
 //
-exports.connect = exports.createConnection = function() {
-  var args = new Array(arguments.length);
-  for (var i = 0; i < arguments.length; i++)
-    args[i] = arguments[i];
-  args = normalizeArgs(args);
-  debug('createConnection', args);
-  var s = new Socket(args[0]);
+exports.connect = exports.createConnection = function(...args) {
```

Comment:
Related: https://fhinkel.github.io/six-speed/#test-rest says rest parameters are faster, but our restparams-bench benchmark says otherwise...though none of them are testing overloaded arguments